linera-storage-service/src/client.rs

```diff
@@ -351,23 +352,40 @@ impl ServiceStoreClientInternal {
         }
     }
 
+    async fn read_single_entry(
+        &self,
+        message_index: i64,
+        index: i32,
+    ) -> Result<Vec<u8>, ServiceStoreError> {
+        let channel = self.channel.clone();
+        let query = RequestSpecificChunk {
+            message_index,
+            index,
+        };
+        let request = tonic::Request::new(query);
+        let mut client = StoreProcessorClient::new(channel);
```

Comment:
Can't we reuse the client between reads?

Reply:
No:
* The reads are done simultaneously so no reuse.
* Using the same client between reads was the old code.

The recommended way of doing queries is to clone channels. See https://docs.rs/tonic/latest/tonic/transport/struct.Channel.html where it is said, "Due to this cloning the Channel type is cheap and encouraged".